comp/metadata/packagesigning/packagesigningimpl/aptsigning.go

```diff
@@ -0,0 +1,185 @@
+					decryptGPGFile(allKeys, repoFile{name, repos}, "signed-by", client, logger)
+				}
+			}
+		}
+	}
+}
+
+func parseSourceListFile(filePath string) map[string][]pkgUtils.Repositories {
+	reposPerKey := make(map[string][]pkgUtils.Repositories)
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+		splitLine := sourceListRegexp.FindStringSubmatch(line)
+		if len(splitLine) > 1 {
+			options := splitLine[2]
+			keyURI := signedBy.FindStringSubmatch(options)
+			if len(keyURI) > 1 {
+				if _, ok := reposPerKey[keyURI[1]]; !ok {
+					reposPerKey[keyURI[1]] = []pkgUtils.Repositories{{RepoName: strings.ReplaceAll(splitLine[3], " ", "/")}}
+				} else {
+					reposPerKey[keyURI[1]] = append(reposPerKey[keyURI[1]], pkgUtils.Repositories{RepoName: strings.ReplaceAll(splitLine[3], " ", "/")})
+				}
+			}
+		}
+	}
+	return reposPerKey
+}
+
+func getDebsigKeyPaths() []string {
+	filePaths := make(map[string]struct{})
+	// Search in the policy files
+	if _, err := os.Stat(debsigPolicies); err == nil {
+		if debsigDirs, err := os.ReadDir(debsigPolicies); err == nil {
+			for _, debsigDir := range debsigDirs {
+				if debsigDir.IsDir() {
+					if policyFiles, err := os.ReadDir(filepath.Join(debsigPolicies, debsigDir.Name())); err == nil {
+						for _, policyFile := range policyFiles {
+							// Get the gpg file name from policy files
+							if debsigFile := getDebsigFileFromPolicy(filepath.Join(debsigPolicies, debsigDir.Name(), policyFile.Name())); debsigFile != "" {
+								debsigFilePath := filepath.Join(debsigKeyring, debsigDir.Name(), debsigFile)
+								if _, err := os.Stat(debsigFilePath); err == nil {
+									filePaths[debsigFilePath] = struct{}{}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	// Denormalise the map
+	filePathsSlice := make([]string, 0)
+	for k := range filePaths {
+		filePathsSlice = append(filePathsSlice, k)
+	}
+	return filePathsSlice
+}
+
+// policy structure to unmarshall the policy files
+type policy struct {
+	XMLName      xml.Name `xml:"Policy"`
+	Origin       origin   `xml:"Origin"`
+	Selection    selection
+	Verification verification
+}
+
+type origin struct {
+	Name        string `xml:"Name,attr"`
+	ID          string `xml:"id,attr"`
+	Description string `xml:"Description,attr"`
+}
+
+type selection struct {
+	Required required `xml:"Required"`
+}
+
+type required struct {
+	Type string `xml:"Type,attr"`
+	File string `xml:"File,attr"`
+	ID   string `xml:"id,attr"`
+}
+
+type verification struct {
+	MinOptional int      `xml:"MinOptional,attr"`
+	Required    required `xml:"Required"`
+}
+
+func getDebsigFileFromPolicy(policyFile string) string {
+	if xmlData, err := os.ReadFile(policyFile); err == nil {
+		var policy policy
+		if err = xml.Unmarshal(xmlData, &policy); err == nil {
+			return policy.Verification.Required.File
```

Comment:
We seem to only care about `policy.Verification.Required.File`; still, we are mapping other values to the policy variable. Do we need to do that?

Reply:
I had the full struct auto-generated by copilot :) I don't know if there is a real gain in ignoring the other fields of the xml, wdyt?

Reply:
> I had the full struct auto-generated by copilot :)

Nice ©️ 

> I don't know if there is a real gain in ignoring the other fields of the xml, wdyt?

That works for me. I usually tend only to map the information I'm using. I'm not sure either if there is any actual gain from not mapping all values.